discussions/prog_repo.md

```diff
@@ -0,0 +1,192 @@
+
+We recognize that it may make sense to migrate the eBPF Package Repository
+(eBPF Programs within the repository) out of the L3AF Project and into its own
+Linux Foundation project in the future. Doing this initially may not make sense (due to L3AF-specific
+eBPF program chaining implementation for networking programs), but as L3AF and other projects mature,
+a platform-agnostic repository could be useful for multiple projects.
+
+# What should an initial version look like
+
+This section examines simple ways to create a location where eBPF Programs can be uploaded
+and made available for others to download.
+
+## A GitHub repository may be sufficient
+
+We would like to leverage GitHub:
+
+- A new GitHub repository will be created to store eBPF program source code
+- All submissions will be manually reviewed by the L3AF team
+- Once approved, programs will be published in the eBPF Package Repository
+
+Another important thing to note is that, initially, code submissions will need
+to conform to L3AF's eBPF program chaining mechanics.
+
+## To Build or Not to Build
+
+Because of the overhead and support requirements of a full build system, it may not be
+feasible to build the eBPF program source code in the initial version. However, we believe
+this would greatly hinder adoption from both contributors and users. Our proposal, therefore,
+is to have the repository’s initial version include scripts (e.g. Dockerfile for build system images),
+and steps to build eBPF Programs locally.
+
+## eBPF Package Repository of the future
+
+Future versions could build on top of the foundation laid by the initial version.
+
+# What should a future version look like
+
+This section examines simple ways to create a location where eBPF Programs from trusted parties
+can be uploaded and made available for others to download. This can also enable users to provide
+rating, reviews, and tags to the packages.
+
+### Build Process
+
+Contributed eBPF programs can be built on common images for Linux and other platforms.
+Contributors are expected to build their code using the most recent image used by our build system.
+Contributors could have a choice to build on Linux only or other platforms.
+
+Build artifacts (i.e., eBPF program bytecode and user space binaries) could be stored
+in a public file storage repository. Users could download the artifacts directly
+from this repository. In fact, such a repository could also be considered the
+eBPF Package Repository, and the source repository could be a separate entity.
+
+### Portability
+
+It is hard to discuss a build system without broaching the topic of portability.
+The eBPF portability story has improved recently with the introduction of eBPF CO-RE on Linux.
+Previous issues with portability and the eBPF CO-RE solution are explained here:
+
+https://nakryiko.com/posts/bpf-portability-and-co-re/
+
+Because an eBPF Package Repository is a place from where contributed eBPF programs (byte code)
```

Comment:
As far as we currently know, byte code won't be sufficient for cases like hypervisor-enforced code integrity.  For such cases, the package would need to contain native code not ebpf byte code.

Reply:
@dthaler We can explore this use case going forward.

Reply:
Perhaps we can remove the (byte code) designation here and say while compiled is the target, we can leave the door open to interpreted (native) code in the eBPF package repo